### ISSUE TITLE
Migrate Catalog Browser to API Wrapper

### ISSUE THREAD

**Issue by SebastianOpriel** [sovity/edc-ui#232]
# Feature
## Description
Right now the catalog browser does multiple requests in the UI and cannot properly include further information for contract offers such as existing negotiations or existing contract agreements

## Possible Solution

- [ ] In general, this is on hold until our broker server / broker client have been implemented further so we can judge how the broker and the new catalog browser will interact.

### MVP
- [ ] Create API Wrapper Endpoint (MVP endpoint, only include asset for each contract offer)
  - [ ] Accepts a list of additional Connector Endpoints
  - [ ] Has a flag to skip preconfigured Connector Endpoints
  - [ ] Requests Connector Endpoints in parallel
  - [ ] Returns a status for the entire request, in case all requests failed.
  - [ ] Returns a list of request status results, e.g. for each url if it succeeded, and if it didn't, a comprehensible user error message
  - [ ] Returns a merged list of contract offer dtos.
  - [ ] Accept search parameters, search should happen on the backend side.
- [ ] Migrate the UI to use the new endpoint. This should allow us to remove existing logic about merging results / merging status results

### Potential Improvements (to be implemented later)
- [ ] #211 
- [ ] Existing Contract Agreement details
- [ ] Transfer History for each contract offer

**Comment by richardtreier** [comment]
replaced by #403
